### PR TITLE
Extract dedup vid request data logics

### PR DESCRIPTION
**Pull request by yixinglu**
GetVertices and GetNeighbors share the same codes with cpp template partial specialization

### CODE REVIEW

File: src/executor/StorageAccessExecutor.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (yixinglu) [inline src/executor/StorageAccessExecutor.cpp:1]: 2021

File: src/executor/StorageAccessExecutor.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+DataSet StorageAccessExecutor::buildRequestDataSet(Iterator *iter, Expression *expr, bool dedup) {
+    DCHECK(iter && expr) << "iter=" << iter << ", expr=" << expr;
+    nebula::DataSet vertices({kVid});
+    vertices.rows.reserve(iter->size());
+
+    std::unordered_set<ValueType> uniqueSet;
+    uniqueSet.reserve(iter->size());
+
+    const auto &spaceInfo = qctx()->rctx()->session()->space();
+    auto vidType = spaceInfo.spaceDesc.vid_type;
```
Comment (yixinglu) [inline src/executor/StorageAccessExecutor.cpp:53]: reference

File: src/executor/StorageAccessExecutor.h [hunk 3]
```diff
@@ -125,6 +131,13 @@ class StorageAccessExecutor : public Executor {
                 folly::stringPrintf("%d(us)/%d(us)", std::get<1>(info), std::get<2>(info)));
         }
     }
+
+    bool isIntVidType() const;
+
+    template <typename ValueType>
```
Comment (yixinglu) [inline src/executor/StorageAccessExecutor.h:137]: rename to vidtype

File: src/executor/StorageAccessExecutor.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    std::unordered_set<VidType> uniqueSet;
+    uniqueSet.reserve(iter->size());
+
+    const auto &vidType = space.spaceDesc.vid_type;
+
+    for (; iter->valid(); iter->next()) {
+        auto vid = expr->eval(exprCtx(iter));
+        if (!SchemaUtil::isValidVid(vid, vidType)) {
+            LOG(WARNING) << "Mismatched vid type: " << vid.type()
```
Comment (laura-ding) [inline src/executor/StorageAccessExecutor.cpp:54]: here can't only print log, need to return an error to users.

**Review by laura-ding — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**